docs/advanced/middleware.md

```diff
@@ -0,0 +1,32 @@
+In most cases the `dispatcher` in our `store` sends incoming `action` objects directly to the `reducer` to determine what updates should be made to the `state` object tree, but sometimes we want some additional processing to be done on the `action` that our `reducer` cannot provide. Features like delayed processing of an `action`, logging `action` objects dispatched to our `store`, or waiting for the response from a network request before sending our `action` payload to the `reducer` are quite cumbersome within the strict confines of Rodux. We circumvent this problem by introducing custom `middleware` that processes `action` objects before they are passed on to the `reducer`, allowing us to break out of the Rodux paradigm in a controlled and consistent fashion when necessary.
+
+When creating a `store` via [`Store.new`](../api-reference.md#storenew), you can optionally provide `middleware` like so:
+
+!!! info
+	the following examples assume that you've successfully [installed Rodux](installation.md) into `ReplicatedStorage`!
+
+```lua
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+
+local Rodux = require(ReplicatedStorage.Rodux)
+
+local reducer = function(state, action)
+	-- the body of your reducer
+end
+
+local initialState = {}
+
+local delayOneSecondMiddleware = function(nextDispatch, store)
+	return function(action)
+		delay(1, function()
+			nextDispatch(action)
+		end)
+	end
+end
+
+local store = Rodux.Store.new(reducer, initialState, {
+	delayOneSecondMiddleware,
+})
+```
+
+Rodux has two `middlewares` available to you out of the box. See [`Thunks`](thunks.md) and [`Debugging`](../debugging.md) for more details.
```

Comment:
Right, my main complaint was that you said "there are out-of-the-box middleware" but then you linked to other pages that talked about orthogonal things that USED that middleware, rather than sending them directly to an explanation of what those middleware are.  I think linking directly to the section for of out-of-the-box middleware in the api does what we're both looking for 👍